### DISCUSSION TITLE
Where do I put the provision profile on the ADT command line for macOS app

### DISCUSSION THREAD

**Discussion by danielkotzer** [airsdk/Adobe-Runtime-Support#2612] [General]
I have successfully created a macOS application on my mini mac with ADT using a *.command file I can simply double click to run, here is its content:
#!/bin/bash cd "$(dirname "$BASH_SOURCE")" || { echo "Error getting script directory" >&2 exit 1 } bash /Applications/AIRSDK_MacOS/bin/adt -package -storetype KeychainStore -alias "3rd Party Mac Developer Application: MyCompany, Inc. (xxxxxxxxxx)" -target bundle  -storetype KeychainStore -alias "3rd Party Mac Developer Application: MyCompany, Inc. (xxxxxxxxxx)" AppName.app AppName-app.xml -extdir ane AppName.swf assets icons

then I sign it with another command and create a pkg file and upload it to mac app store successfully with Transporter, problem is I get one issue while uploading: the app is missing a provision profile which makes it impossible to use it for testflight.

I try to add this line to the command line but all I get is errors:
-provisioning-profile AppName_Mac_Store.provisionprofile

Where and how should I add it to the command line shown above?

**Comment by ajwfrost** [comment]
Hi

There's no specific provisioning profile support for macOS in ADT - I don't believe we need this, but if you're not able to get this working using the below suggestions, we can look at adding it!

But if you can rename your provisioning profile to "embedded.mobileprovision" and then add this into the list of files for packaging up, alongside your SWF file (so `... -extdir ane AppName.swf embedded.mobileprovision assets icons`), then it should be stored in the right place..

thanks

**Comment by danielkotzer** [comment]
It didn't work, it did create the app but when uploading it with Transporter I got the same message:
"Cannot be used with TestFlight because the bundle at “AppName.app” is missing a provisioning profile. Main bundles are expected to have provisioning profiles in order to be eligible for TestFlight." (90889)

**Comment by danielkotzer** [comment]
Need help on this issue, I have an app I uploaded to the mac store and need to make the in app purchasing work.
What am I supposed to do?

**Comment by ajwfrost** [comment]
Hi - we've got a change in the ADT tool coming out shortly that will support this for macOS app bundles; the challenge at the moment is that the file needs to be put into the "Contents" location in the bundle, but all app-specific files (SWFs etc) that you specify on the command line are put into "Contents/Resources".

The new release should be available on Monday, assuming everything goes well.. (if it's very urgent, we could just send you a patch file for the adt.jar?)

thanks

**Comment by danielkotzer** [comment]
Not urgent, will wait for the new releas  which will be what version?

בתאריך יום ד׳, 17 במאי 2023, 22:50, מאת Andrew Frost ‏<
***@***.***>:

**Reply by ajwfrost** [reply]
Okay thanks.. All the changes we've made are essentially bug fixes, so it would be 50.2.2.6.

**Comment by danielkotzer** [comment]
you wrote:
"Hi - we've got a change in the ADT tool coming out shortly that will
support this for macOS app bundles; the challenge at the moment is that the
file needs to be put into the "Contents" location in the bundle, but all
app-specific files (SWFs etc) that you specify on the command line are put
into "Contents/Resources"."

Does this mean I have to put the provision profile into the bundle after
creating it, opening it's content folder and putting it mauly into the
contents folder?

**Reply by ajwfrost** [reply]
If you're using 50.2.2.6, then you can just add the `-provisioning-profile` argument on the ADT command line.

Otherwise, yes the manual process would be to put the provisioning profile manually into the bundle and then code-sign the whole thing again..
